chacha20/src/lib.rs

```diff
@@ -16,13 +16,14 @@ use zeroize::Zeroize;
 use salsa20_core::{SalsaFamilyCipher, SalsaFamilyState};
 
 /// Wrapper for state for ChaCha-type ciphers.
-struct ChaChaState {
+struct ChaChaState<N: ArrayLength<u8>> {
     state: SalsaFamilyState,
+    phantom: core::marker::PhantomData<N>,
 }
 
 /// The ChaCha20 cipher.
-pub struct ChaCha20 {
-    state: ChaChaState,
+pub struct ChaCha20<N: ArrayLength<u8>> {
```

Comment:
This seems like it could benefit from a default so most users can ignore the generic parameter, e.g.:

```rust
pub struct ChaCha20<N: ArrayLength<u8> = U8> { ... }
```

What the default should be seems debatable. I'll note one of the CFRG work items is XChaCha20, which derives a unique key (and associated ChaCha20 nonce) using HChaCha20, and by doing so provides additional security in the multi-user setting and would be my general preference for a "wide nonce" to recommend to end users:

https://datatracker.ietf.org/doc/draft-arciszewski-xchacha/